tools/system_libs.py

```diff
@@ -1724,6 +1724,57 @@ def variations(cls):
             [dict(malloc='emmalloc-verbose', **combo) for combo in combos if combo['verbose'] and not combo['memvalidate']])
 
 
+class libmimalloc(MTLibrary):
+  name = 'libmimalloc'
+
+  cflags = [
+    '-fno-builtin',
+    '-Wno-deprecated-pragma',
+    # build emmalloc as only a system allocator, without exporting itself onto
+    # malloc/free in the global scope
+    '-DEMMALLOC_NO_STD_EXPORTS',
+    # build mimalloc with an override of malloc/free
+    '-DMI_MALLOC_OVERRIDE',
+    # TODO: add build modes that include debug checks 1,2,3
+    '-DMI_DEBUG=0',
+  ]
+
+  # malloc/free/calloc are runtime functions and can be generated during LTO
+  # Therefor they cannot themselves be part of LTO.
+  force_object_files = True
+
+  includes = ['system/lib/mimalloc/include']
+
+  # build all of mimalloc, and also emmalloc which is used as the system
+  # allocator underneath it.
+  def get_files(self):
+    return files_in_path(
+      path='system/lib/mimalloc/src',
+      filenames=[
+        'alloc.c',
+        'alloc-aligned.c',
+        'alloc-posix.c',
+        'arena.c',
+        'bitmap.c',
+        'heap.c',
+        'init.c',
+        'options.c',
+        'os.c',
+        'page.c',
+        'random.c',
+        'segment.c',
+        'segment-map.c',
+        'stats.c',
+        'prim/prim.c',
```

Comment:
Isn't this all the `.c` files?  Use a glob instead?

Reply:
It's not all files, e.g, `alloc-override.c` is not in this list (`mimalloc includes it itself at the source level, from `alloc.c`). So we do need this manual list I'm afraid.

Reply:
Ok, done in

https://github.com/emscripten-core/emscripten/pull/20651/commits/879d9b9feb6b8e214424a9da248db72dfa846853

Still kind of messy. Do you prefer that?